tls/s2n_connection.c

```diff
@@ -1033,11 +1033,13 @@ const char *s2n_connection_get_curve(struct s2n_connection *conn)
 {
     PTR_ENSURE_REF(conn);
 
-    if (!conn->secure.server_ecc_evp_params.negotiated_curve) {
-        return "NONE";
+    if (conn->secure.server_ecc_evp_params.negotiated_curve) {
+        if (conn->actual_protocol_version >= S2N_TLS13 || s2n_kex_includes(conn->secure.cipher_suite->key_exchange_alg, &s2n_ecdhe)) {
```

Comment:
I *think* this line only works as long as we only support ECC groups for TLS1.3. Worth a comment here or a unit test that fails once a non ECC group is added to s2n?

Reply:
Hmmm any ideas how to write that unit test? IFAIK there's no list of groups, its just ECC groups or KEM groups taken from security policies.

Reply:
Since this is an improvement over the previous behavior, happy to not make it a blocker. Shall we track that in an Issue to increase likelihood it is addressed before we add non ECC groups in the future?

Reply:
Added comment and issue #2924